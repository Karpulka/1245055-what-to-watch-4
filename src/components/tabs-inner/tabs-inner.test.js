import React from "react";
import renderer from "react-test-renderer";
import TabsInner from "./tabs-inner";

const overview = {
  title: `Overview`,
  value: {
    description: `This is Description. TCHK.`,
    rating: 8.5,
    voiceCount: 214,
    director: `And he's Director`,
    actorList: [`Actor`, `Actor`, `Actor`]
  }
};

const details = {
  title: `Details`,
  value: {
    genre: `Drama`,
    year: 2012,
    director: `He's Director`,
    actorList: [`Actor 1`, `Actor 2`],
    runtime: 132
  }
};

const reviews = {
  title: `Reviews`,
  value: 1
};

describe(`Render Tabs Variants`, () => {
  it(`Render TabsInner Overview`, () => {
    const tree = renderer
      .create(<TabsInner tabValue={overview}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render TabsInner Details`, () => {
    const tree = renderer
      .create(<TabsInner tabValue={details}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render TabsInner Reviews`, () => {
    const tree = renderer
      .create(<TabsInner tabValue={reviews}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
