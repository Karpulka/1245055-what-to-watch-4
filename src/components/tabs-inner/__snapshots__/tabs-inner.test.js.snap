// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Tabs Variants Render TabsInner Details 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        He's Director
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Actor 1
        <br />
        Actor 2
        <br />
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        2h 12m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Drama
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2012
      </span>
    </p>
  </div>
</div>
`;

exports[`Render Tabs Variants Render TabsInner Overview 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        214
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    This is Description. TCHK.
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        And he's Director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Actor, Actor, Actor
         and other
      </strong>
    </p>
  </div>,
]
`;

exports[`Render Tabs Variants Render TabsInner Reviews 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        6.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.2
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.7
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        6.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.2
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.7
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        6.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.2
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.7
      </div>
    </div>
  </div>
</div>
`;
